### PR TITLE
Add efficient sequence utilization

### PR DESCRIPTION
**Pull request by Adefioye**
Here in this PR, the [issue 6](https://github.com/chrisjmccormick/shared-subspaces/issues/6) is being resolved. The goal is to implement efficient sequence packing in order to increase performance of the encoder. This was achieved by adding `enable_packing` property in the `config files` with the goal of helping to decide whether to use packing or not. We expect an improvement because there is more computation spend on real tokens(instead of padding tokens) in the forward/backward pass thereby providing quality gradient signals for learning on training data.

Experiments show that sequence packing helps to improve performance of encoders on pre-training and downstream tasks as shown in table below.

| Model | Metric        | Before  | After   | Δ (diff)  |
|-------|---------------|---------|---------|-----------|
| MLA   | MLM Accuracy  | 55.51%  | 56.51%  | +1.00%    |
| MLA   | SST Accuracy  | 82.34%  | 84.17%  | +1.83%    |
| MLA-O | MLM Accuracy  | 54.61%  | 55.63%  | +1.02%    |
| MLA-O | SST Accuracy  | 83.72%  | 84.17%  | +0.45%    |

Findings:
1. After sequence packing, the MLM accuracy and SST accuracy improved for both MLA and MLA-O.
2. There was no improvement in the training speed for both architectures. However, the performance on both architectures were about the same on SST at about 84.17%.

### CODE REVIEW

**Review by chrisjmccormick — APPROVED** [review]
Looks great!
